r2/store.go

```diff
@@ -25,11 +25,11 @@ import "github.com/m3db/m3metrics/rules"
 // Store is a construct that can perform operations against a backing rule store.
 type Store interface {
 	// Fetch namespaces and their version
-	FetchNamespaces() ([]string, int, error)
+	FetchNamespaces() (*rules.NamespacesView, error)
 
-	CreateNamespace(namespaceID string) (string, error)
+	CreateNamespace(namespaceID string, author string) (*rules.NamespaceView, error)
```

Comment:
Might be good to take an options struct here in case we need more information to be added so it doesn't break the API. Same applies to the rest of the places where we pass in the `author` explicitly.